grails-app/controllers/au/org/ala/biocache/hubs/OccurrenceController.groovy

```diff
@@ -426,4 +428,54 @@ class OccurrenceController {
         combined.compareRecord = webServicesService.getCompareRecord(id)
         render combined as JSON
     }
+
+    /**
+     * Find any cl1234 or el1234 layer IDs in the q or fq params and return a list of layer IDs
+     *
+     * @param searchRequestParams
+     * @return List
+     */
+    List getListOfLayerIds(SpatialSearchRequestParams searchRequestParams) {
+        List layersList = [] // list of IDs
+        List queries = [] // list of q and fq query strings
+        queries.add(searchRequestParams.q) // q param
+        queries.addAll(searchRequestParams.fq) // array of fq params
+        log.debug "queries = ${queries}"
+
+        queries.each { String q ->
+            // regex for find el1234 and cl5678 IDs
+            Matcher layer = q =~ /^\-*([ec]l\d+)\:.*/
+
+            if (layer.matches()) {
+                String match = layer[0][1] // grab the captured text
+
+                if (match && match.length() > 3) {
+                    layersList.add(match)
+                }
+            }
+        }
+
+        layersList
+    }
+
+    /**
+     * For a given list of layer IDs (el1234 or cl456) do a lookup against spatial service
+     * for metadata for the layer ID and return a list of layer objects.
+     *
+     * @param layerIdsList
+     * @return List
+     */
+    List getListOfLayerObjects(List layerIdsList) {
```

Comment:
These methods could be private or put in a service unless they're supposed to be called as web services at some point?

Reply:
Yes, should be in service - groovy doesn't respect `private` keyword last time i checked. I'll change and update PR.